homeassistant/components/androidtv/__init__.py

```diff
@@ -1 +1,214 @@
 """Support for functionality to interact with Android TV/Fire TV devices."""
+import json
+import logging
+import os
+
+from adb_shell.auth.keygen import keygen
+from androidtv import state_detection_rules_validator
+from androidtv.adb_manager.adb_manager_sync import ADBPythonSync
+from androidtv.setup_async import setup
+
+from homeassistant.components.media_player import DOMAIN as MP_DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_HOST,
+    CONF_PORT,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.storage import STORAGE_DIR
+
+from .const import (
+    ANDROID_DEV,
+    ANDROID_DEV_OPT,
+    CONF_ADB_SERVER_IP,
+    CONF_ADB_SERVER_PORT,
+    CONF_ADBKEY,
+    CONF_STATE_DETECTION_RULES,
+    DEFAULT_ADB_SERVER_PORT,
+    DEVICE_ANDROIDTV,
+    DEVICE_FIRETV,
+    DOMAIN,
+    PROP_SERIALNO,
+    SIGNAL_CONFIG_ENTITY,
+)
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+RELOAD_OPTIONS = [CONF_STATE_DETECTION_RULES]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _setup_androidtv(hass, config):
+    """Generate an ADB key (if needed) and load it."""
+    adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))
```

Comment:
Is it a problem if multiple devices are using the same key?

Reply:
No, as I now. I have 2 FireTV and they use same key.